test/app-decorator.js

```diff
@@ -148,6 +148,83 @@ test('operationName', async (t) => {
   })
 })
 
+test('replaceSchema with makeSchemaExecutable', async (t) => {
+  const app = Fastify()
+
+  app.register(GQL, {
+    schema: makeExecutableSchema({
+      typeDefs: `
+      type Query {
+        add(x: Int, y: Int): Int
+      }
+    `,
+      resolvers: {
+        Query: {
+          add: async (_, { x, y }) => x + y
+        }
+      }
+    })
+  })
+
+  app.register(async function (app) {
+    app.graphql.replaceSchema(
+      makeExecutableSchema({
+        typeDefs: `
+        type Query {
+          add(x: Int, y: Int, z: Int): Int
+        }
+      `,
+        resolvers: {
+          Query: {
+            add: async (_, { x, y, z }) => x + y + z
+          }
+        }
+      })
+    )
+  })
+
+  // needed so that graphql is defined
+  await app.ready()
+
+  const query = '{ add(x: 2, y: 2, z: 2) }'
+  const res = await app.graphql(query)
+
+  t.deepEqual(res, {
+    data: {
+      add: 6
+    }
+  })
+})
```

Comment:
Can you add a test where you verify that the caches are cleared? What you need is to issue the query once (it gets cached), then _remove_ that part of the schema, and then issue the same query again. The second call should give an error.

Reply:
I've added a test for checking cache.
At the first call, we have two queries (`add` and `subtract`), I execute `subtract` and checking output.
At the second call, we have one query (`add`), I execute `subtract` and checking error.